pkg/nfd-master/nfd-master.go

```diff
@@ -876,6 +907,12 @@ func (m *nfdMaster) nfdAPIUpdateOneNode(cli k8sclient.Interface, node *corev1.No
 		return fmt.Errorf("failed to merge NodeFeature objects for node %q: %w", node.Name, err)
 	}
 
+	// If we deny node feature labels, we'll empty the labels variable
+	if m.config.Restrictions.DenyNodeFeatureLabels {
+		klog.V(2).InfoS("node feature labels are disabled in configuration (restrictions.denyNodeFeatureLabels=true)")
+		nodeFeatures.Labels = map[string]string{}
```

Comment:
As per documentation (in `docs/reference/master-configuration-reference.md`) we should disable/deny labels from 3rd party NFs. So the one created by nfd-worker should be ok. We need to move this check to `getAndMergeNodeFeatures()`. Then, how to determine the NF created by nfd-worker: I think the logic would be namespace==our namespace AND name==nodeName.

Thoughts?